### PR TITLE
added the ability to transpile from tsx (typescript + react)

### PR DESCRIPTION
**Pull request by mgebundy**
## Description
Adding Typescript support to React > Vue compiler

## Motivation and Context
Makes for cleaner React components!

## How Has This Been Tested?
Converted a test component (`Button.jsx`) to a Typescript + React component in order to test the compiler

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] I have included a vue-play example (if this is a new feature)

### CODE REVIEW

File: tests/data/components/Button/index.tsx [hunk 1]
```diff
@@ -63,7 +62,19 @@ const blink = keyframes`
     }
 `;
 
-const Dot = styled.div`
+interface DotProps {
+	dotSize: {
+		small: number,
```
Comment (quinnlangille) [inline tests/data/components/Button/index.tsx:67]: indentation is off here

File: tests/data/components/Button/index.tsx [hunk 2]
```diff
@@ -75,15 +86,24 @@ const Dot = styled.div`
...
+	animation: ${({ dotCount, blinkInterval, index }) =>
+		`${dotCount * blinkInterval}s ${(blinkInterval / 2) *
 			index}s ${blink} step-end infinite`};
 `;
 
 /**
  * Button Description
  */
-class Button extends Component {
+class Button extends Component<{
```
Comment (quinnlangille) [inline tests/data/components/Button/index.tsx]: This PR is standard setting so I wonder if we should experiment a bit here. I don't particularly like the in line type argument, but I don't know if react allows us to use an interface or declared type here. Thoughts on other ways of declaring the types?

**Review by quinnlangille — COMMENTED** [review]
Looks rly cool! Lets discuss how we can better implement typescript as a new standard for the repo - should we lean more towards functional components (I don't know if the compiler can even handle those currently, would have to test)? What alternatives for inline typing do we have with react-typescript?

File: tests/data/components/Button/index.tsx (hunk 1 above)
Comment (mgebundy) [reply tests/data/components/Button/index.tsx:67]: looks fine in my ide :|

File: tests/data/components/Button/index.tsx (hunk 2 above)
Comment (mgebundy) [reply tests/data/components/Button/index.tsx]: Pulled them out into interfaces

**Review by quinnlangille — APPROVED** [review]

**Comment by mgebundy** [comment]
Updated the typings in this PR

I'll work on a test with a functional component, the library should be able to handle those

**Pull request merged**